api/rest/bundle_handler.go

```diff
@@ -69,7 +77,20 @@ type BundleHandler struct {
 	clock                 Clock
 	workDir               string                // location where bundles are generated and stored
 	collectors            []collector.Collector // information what should be in the bundle
+	coordinator           Coordinator           // coordinator for building remote bundles
 	bundleCreationTimeout time.Duration         // limits how long bundle creation could take
+	urlBuilder            dcos.NodeURLBuilder
+}
+
+type createArguments struct {
+	BundleType string `json:"type"`
+	Nodes      []node `json:"nodes"`
+}
+
+type node struct {
+	IP      net.IP `json:"ip"`
+	Role    string `json:"role"`
```

Comment:
When do we serialize `node` to JSON?

Reply:
We serialize it from json. This is a change to the API in order to get the nodes' roles though maybe we could pull this from within the cluster using the IP alone.